src/wp-includes/admin-bar.php

```diff
@@ -1231,10 +1231,7 @@ function wp_admin_bar_add_secondary_groups( $wp_admin_bar ) {
  * @since 3.1.0
  */
 function wp_admin_bar_header() {
-	$type_attr = current_theme_supports( 'html5', 'style' ) ? '' : ' type="text/css"';
-	?>
-<style<?php echo $type_attr; ?> media="print">#wpadminbar { display:none; }</style>
-	<?php
+	wp_add_inline_style( 'admin-bar', ' @media print { #wpadminbar { display:none; } }' );
```

Comment:
The big downside of the removal of `<style>` is that the CSS no longer gets syntax highlighting and IDE inspections. This can be mitigated a bit in PhpStorm specifically via [language injection comments](https://www.jetbrains.com/help/phpstorm/using-language-injections.html#use-language-injection-comments)

```suggestion
	wp_add_inline_style( 'admin-bar', /* language=CSS */ '@media print { #wpadminbar { display:none; } }' );
```

Before:

![image](https://github.com/WordPress/wordpress-develop/assets/134745/c4561c49-798a-4179-abbf-bb54b85a10c6)


After:

![image](https://github.com/WordPress/wordpress-develop/assets/134745/e520f3fb-3c02-43db-baa6-96188b95c4d6)

However, these don't work in VSCode. See [comment](https://core.trac.wordpress.org/ticket/58664#comment:15) on Core-58664 regarding this same issue for JS, with a workaround to retain the IDE support. That comment also links off to a Slack thread.